src/autodiff.jl

```diff
@@ -319,68 +382,135 @@ function uncompress_kernel!(J::CUSPARSE.CuSparseMatrixCSR, compressedJ, coloring
 end
 
 """
-    residual_jacobian!(arrays::StateJacobian,
+    residual_jacobian!(J::Jacobian,
                         residual_polar!,
-                        v_m, v_a, ybus_re, ybus_im, pinj, qinj, pv, pq, ref, nbus,
-                        timer = nothing)
+                        vm, va, ybus_re, ybus_im, pinj, qinj, pv, pq, ref,
+                        nbus, type::AbstractJacobian)
 
 Update the sparse Jacobian entries using AutoDiff. No allocations are taking place in this function.
 
-* `arrays::StateJacobian`: Factory created Jacobian object to update
+* `J::Jacobian`: Factory created Jacobian object to update
 * `residual_polar`: Primal function
-* `v_m, v_a, ybus_re, ybus_im, pinj, qinj, pv, pq, ref, nbus`: Inputs both
+* `vm, va, ybus_re, ybus_im, pinj, qinj, pv, pq, ref, nbus`: Inputs both
   active and passive parameters. Active inputs are mapped to `x` via the preallocated views.
-
+* `type::AbstractJacobian`: Either `StateJacobian` or `ControlJacobian`
 """
-function residual_jacobian!(arrays::Jacobian,
+function residual_jacobian!(J::Jacobian,
```

Comment:
This renaming is nice!

Reply:
Thanks. I think this was one of the oldest untouched code sections. It still needs more work.